server/pkg/layer/infobox_field.go

```diff
@@ -51,3 +52,10 @@ func (i *InfoboxField) ValidateProperty(pm property.Map) error {
 
 	return nil
 }
+
+func (i *InfoboxField) Upgrade(id plugin.ID) {
+	if i == nil {
```

Comment:
```suggestion
	if i == nil || i.plugin.Name() != id.Name() {
```

Reject to avoid upgrading to a completely unrelated plugin.